### PR TITLE
Remove the duplication nodes with the outline explorer

### PR DESCRIPTION
**Pull request by jdneo**
Screen Shot 2020-02-13 at 10 45 49 AM

This is the first step of #229: Remove the duplication in the dependency explorer with the outline explorer, which containing the following change:

- Remove the ITypeRoot level of in the dependency explorer. Instead, the explorer will directly show the primary class under the packages.
- Change the configuration `showOutline` to `showMembers`, which can specify whether to show the class members under the primary class, so:
  - The package declaration will not show now, which is duplicated.
  - The primary type itself won't show now.

### CODE REVIEW

File: jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/model/PackageNode.java [hunk 1]
```diff
@@ -187,15 +222,23 @@ public static PackageNode createNodeForClasspathEntry(IClasspathEntry classpathE
...
 
-    public static PackageNode createNodeForTypeRoot(IJavaElement typeRoot) {
-        PackageNode typeRootNode = new TypeRootNode(typeRoot.getElementName(), typeRoot.getPath().toPortableString(), NodeKind.TYPEROOT,
-                typeRoot instanceof IClassFile ? TypeRootNode.K_BINARY : TypeRootNode.K_SOURCE);
-        if (typeRoot instanceof ICompilationUnit) {
-            typeRootNode.setUri(JDTUtils.toURI((ICompilationUnit) typeRoot));
-        } else if (typeRoot instanceof IClassFile) {
-            typeRootNode.setUri(JDTUtils.toUri((IClassFile) typeRoot));
+    public static PackageNode createNodeForPrimaryType(IType type) {
+        PackageNode primaryTypeNode = new PackageNode(type.getElementName(), type.getPath().toPortableString(), NodeKind.PRIMARYTYPE);
```
Comment (testforstephen) [inline jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/model/PackageNode.java]: why create packageNode for IType?

Comment (jdneo) [reply jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/model/PackageNode.java]: That's some unknown historical reason. The Java side code only has the following kinds of nodes:
- PackageNode
- ContainerNode
- PackageRootNode

File: jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java [hunk 2]
```diff
@@ -349,10 +346,23 @@
                 IPackageFragment packageFragment = packageRoot
                         .getPackageFragment(PackageNode.DEFAULT_PACKAGE_DISPLAYNAME.equals(query.getPath()) ? "" : query.getPath());
                 if (packageFragment != null) {
-                    IJavaElement[] types = packageFragment.getChildren();
+                    List<IType> primaryTypes = new ArrayList<>();
+                    for (IJavaElement element : packageFragment.getChildren()) {
+                        if (element instanceof ITypeRoot) {
+                            // Filter out the inner class files
+                            if (element instanceof IClassFile && element.getElementName().contains("$")) {
```
[the remark this answers is not in the document]
Comment (jdneo) [reply jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java]: Yes, the Inner class will become: `XXX$Inner.class` after compilation.

[the remark this answers is not in the document]
Comment (testforstephen) [reply jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java]: Theoretically the java class name itself can use $, but i didn't find a good way to judge whether it's inner class. let's keep the current judge, if there is someone reporting the issue, then fix it.

[the remark this answers is not in the document]
Comment (jdneo) [reply jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java]: ok

**Pull request merged**